config/src/config.rs

```diff
@@ -610,6 +613,18 @@ impl NodeConfig {
             .filter(|network| RoleType::Validator == (&network.role).into())
             .last()
     }
+
+    /// Returns true if network_config is for an upstream network
+    pub fn is_upstream_network(&self, network_config: &NetworkConfig) -> bool {
```

Comment:
but overall not sure if I like interface. Can we have something like `fn upstream_network_id(&self) -> String`

Reply:
Why return the ID if we have the network ID already in the parent function that calls this?